tools/generate_api_docs.py

```diff
@@ -64,6 +64,16 @@
    :nosignatures:
 
    {lowlevel_wrappers}
+
+API Utility Classes
```

Comment:
This was my solution to https://github.com/vega/altair/pull/3427#discussion_r1649699382

Ideally, I'd want to have `when`, `then`, `otherwise` *methods* documented fully and hide `__init__`, `to_dict`.

Currently only the `when` *function* pulls through the docs, which I think isn't helpful alone. 

I'm not sure how to achieve this without changing https://github.com/dangotbanned/altair/blob/54a68a310f48ba9d2201da1820c4abaf459154aa/doc/_templates/class.rst and that would affect all classes (which I'm trying to avoid).

The `narwhals` [docs](https://narwhals-dev.github.io/narwhals/api-reference/expressions/) are an example of what I'd like, but with the 3 classes sharing a single page.